### DISCUSSION TITLE
Signup Requires a Valid password error

### DISCUSSION THREAD

**Discussion by aldrinjenson** [supabase/supabase#5643] [Questions]
Hi, I just want a simple signup  with email and password. I've been trying for hours with the default method specified in the docs, but I'm getting the following error: `Signup Requires a Valid password error`
![image](https://user-images.githubusercontent.com/53407417/155420143-c663d4b2-77ce-41b6-81cd-093fbfb6105f.png)
I tried modifying various settings options  in the auth settings without any luck. Some help to fix this would be really appreciated. Thanks.

**Comment by EdgarsLv** [comment]
try to hardcode password. Maybe you are not correctly passing it with input?

**Comment by chirani** [comment]
Is this solved yet, I am having the same issue, doesn't make sense when you're trying to sign up

**Reply by aldrinjenson** [reply]
can you try giving a password having length greater than 6 or 8?

**Comment by Olyno** [comment]
Hi 👋🏻 It's hard to say with only the error. Check that your data includes an email and a password. Sometimes, the password can be empty, and this error happens when you sign up with an empty password (it already happened to me multiple times).

**Comment by choubari** [comment]
I had the same issue. 
When I hardcoded the password, the next error was that I can't signup with an empty email. I knew that I was getting the request body the wrong way. I'm using NextJS 13.2 with the new route handlers, and this was [the solution](https://stackoverflow.com/a/75656292/12802591) 
![image](https://user-images.githubusercontent.com/56233818/224811246-7efd1b3c-169c-43eb-8b8e-0a0bef68146f.png)
